packages/twenty-server/src/engine/workspace-manager/workspace-cleaner/services/clean.workspace-service.ts

```diff
@@ -0,0 +1,288 @@
+
+      for (const workspaceMembersChunk of workspaceMembersChunks) {
+        await Promise.all(
+          workspaceMembersChunk.map(async (workspaceMember) => {
+            await this.userVarsService.set({
+              userId: workspaceMember.userId,
+              workspaceId: workspace.id,
+              key: USER_WORKSPACE_DELETION_WARNING_SENT_KEY,
+              value: true,
+            });
+
+            await this.sendWarningEmail(
+              workspaceMember,
+              workspace.displayName,
+              daysSinceInactive,
+            );
+          }),
+        );
+      }
+
+      return;
+    }
+  }
+
+  async sendCleaningEmail(
+    workspaceMember: WorkspaceMemberWorkspaceEntity,
+    workspaceDisplayName: string | undefined,
+  ) {
+    const emailData = {
+      inactiveDaysBeforeDelete: this.inactiveDaysBeforeDelete,
+      userName: `${workspaceMember.name.firstName} ${workspaceMember.name.lastName}`,
+      workspaceDisplayName: `${workspaceDisplayName}`,
+    };
+    const emailTemplate = CleanSuspendedWorkspaceEmail(emailData);
+    const html = render(emailTemplate, {
+      pretty: true,
+    });
+    const text = render(emailTemplate, {
+      plainText: true,
+    });
+
+    this.emailService.send({
+      to: workspaceMember.userEmail,
+      bcc: this.environmentService.get('EMAIL_SYSTEM_ADDRESS'),
+      from: `${this.environmentService.get(
+        'EMAIL_FROM_NAME',
+      )} <${this.environmentService.get('EMAIL_FROM_ADDRESS')}>`,
+      subject: 'Your workspace has been deleted',
+      html,
+      text,
+    });
+  }
+
+  async informWorkspaceMembersAndDeleteWorkspace(workspace: Workspace) {
+    const workspaceMembers =
+      await this.userService.loadWorkspaceMembers(workspace);
+
+    this.logger.log(
+      `Sending workspace ${workspace.id} ${
+        workspace.displayName
+      } deletion emails to users ['${workspaceMembers
+        .map((workspaceUser) => workspaceUser.userEmail)
+        .join(', ')}']`,
+    );
+
+    const workspaceMembersChunks = chunk(workspaceMembers, 5);
+
+    for (const workspaceMembersChunk of workspaceMembersChunks) {
+      await Promise.all(
+        workspaceMembersChunk.map(async (workspaceMember) => {
+          await this.userVarsService.delete({
+            userId: workspaceMember.userId,
+            workspaceId: workspace.id,
+            key: USER_WORKSPACE_DELETION_WARNING_SENT_KEY,
+          });
+
+          await this.sendCleaningEmail(workspaceMember, workspace.displayName);
+        }),
+      );
+    }
+
+    await this.workspaceService.deleteWorkspace(workspace.id);
+    this.logger.log(
+      `Cleaning Workspace ${workspace.id} ${workspace.displayName}`,
+    );
+  }
+
+  async batchWarnOrCleanWorkspaces(workspaceIds?: string[]): Promise<void> {
+    this.logger.log(`batchWarnOrCleanWorkspaces running...`);
+
+    const suspendedWorkspaces = await this.workspaceRepository.find({
```

Comment:
I would make workspaceIds mandatory and move this "suspendedWorkpace logic" in the cron side